### PR TITLE
RFC-BYOC-DNNL-Integration

### PR DESCRIPTION
**Pull request by crazydemo**
RFC for OneDNN Integration via BYOC.

### CODE REVIEW

File: rfcs/0069-byoc-onednn-integration.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+- g++ 7
+- 'llvm -mcpu=cascadelake -model=platinum-8280'
+- TVM commitID: 19b23b9
+- MXNet version: V1.8.0
+- OneDNN version: V1.7 / V2.4
+
+Runtime config
+- 20 warm-up and 100 batches
+
+![Figure 1 latency scenario](https://github.com/crazydemo/tvm-rfcs/blob/main/rfcs/assets/latest/latency.png)
```
Comment (areusch) [inline rfcs/0069-byoc-onednn-integration.md:43]: i think you can just use a relative path here and it should render properly. i don't see these images.

File: rfcs/0069-byoc-onednn-integration.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+![Figure 2 Throughput scenario](https://github.com/crazydemo/tvm-rfcs/blob/main/rfcs/assets/latest/throughput.png) 
+
+![Figure 3 Real-time scenario](https://github.com/crazydemo/tvm-rfcs/blob/main/rfcs/assets/latest/real-time.png)
+
+# Reference-level explanation
```
Comment (areusch) [inline rfcs/0069-byoc-onednn-integration.md:49]: could you discuss your test strategy a bit?

**Review by areusch — COMMENTED** [review]
@crazydemo added a couple questions. if you want to discuss this in a high-bandwidth setting, you could add it as an agenda item to the [weekly TVM Community Meeting](https://discuss.tvm.apache.org/t/next-tvm-community-meeting-may-18/12784) (look for a forum post about 3-5 days before the next one).

cc @comaniac @mbs-octoml

File: rfcs/0069-byoc-onednn-integration.md (hunk 2 above)
Comment (areusch) [reply rfcs/0069-byoc-onednn-integration.md:49]: to clarify, here i mean: how will we test this in TVM CI (or can we)?

File: rfcs/0069-byoc-onednn-integration.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+This proposal aims to provide a new approach to integrate oneDNN into TVM via DNNL JSON codegen/runtime by applying the following adjustments to tackle the aforementioned issues: 
+- Register a new “alter_op_layout” function for dnnl to get the optimal layouts for dnnl ops with a new layout auto-query function in Relay.
+- Add a custom pass to rewrite “Conv-Add-Add- ReLu” pattern into “Conv-Add- ReLu” to better handle the pattern comes from BatchNorm Folding (“Conv-bias_add-BN-ReLu”).
```
Comment (comaniac) [inline rfcs/0069-byoc-onednn-integration.md:52]: Can you just leverage MergeComposite to achieve this without introducing a custom pass?

**Review by comaniac — APPROVED** [review]
The RFC LGTM. It basically improves the existing BYOC for OneDNN. I didn't see any compatibility issue or risks.

File: rfcs/0069-byoc-onednn-integration.md (hunk 2 above)
Comment (areusch) [reply rfcs/0069-byoc-onednn-integration.md:49]: cool, i agree the CI is not the place for benchmarking. thanks for adding those.

**Review by areusch — APPROVED** [review]

**Pull request merged**